build.py

```diff
@@ -0,0 +1,368 @@
+        r = plpy.execute("select coalesce(pg_catalog.current_setting('ai.python_lib_dir', true), '{python_install_dir()}') as python_lib_dir")
+        python_lib_dir = r[0]["python_lib_dir"]
+        from pathlib import Path
+        python_lib_dir = Path(python_lib_dir).joinpath(version)
+        import site
+        site.addsitedir(python_lib_dir)
+        from ai import __version__ as ai_version
+        assert(version == ai_version)
+        GD["ai.python_lib_dir"] = python_lib_dir
+    """
+    # trim the leading 4 spaces off each line in the string
+    lines = []
+    for line in inject.splitlines(keepends=True):
+        lines.append(line[4:])
+    inject = "".join(lines)
+    code = input_file.read_text()
+    return code.replace('''ADD-PYTHON-LIB-DIR\n''', inject)
+
+
+def build_sql() -> None:
+    build_sql_control_file()
+    hr = "".rjust(80, "-")
+    osf = output_sql_file()
+    osf.unlink(missing_ok=True)
+    with osf.open("w") as wf:
+        wf.write(f"{hr}\n-- {this_version()}\n\n\n")
+        with sql_dir().joinpath("head.sql").open("r") as rf:
+            shutil.copyfileobj(rf, wf)
+        wf.write("\n\n\n")
+        for inc_file in incremental_sql_files():
+            code = build_incremental_sql_file(inc_file)
+            wf.write(code)
+            wf.write("\n\n\n")
+        for idm_file in idempotent_sql_files():
+            wf.write(f"{hr}\n-- {idm_file.name}\n")
+            wf.write(build_idempotent_sql_file(idm_file))
+            wf.write("\n\n\n")
+        wf.flush()
+        wf.close()
+    for prior_version in prior_versions():
+        dest = sql_dir().joinpath(f"ai--{prior_version}--{this_version()}.sql")
+        dest.unlink(missing_ok=True)
+        shutil.copyfile(osf, dest)
+
+
+def clean_sql() -> None:
+    for f in sql_dir().glob(f"ai--*.*.*--{this_version()}.sql"):
+        f.unlink()
+    sql_dir().joinpath(f"ai--{this_version()}.sql").unlink()
+
+
+def extension_dir() -> Path:
+    if shutil.which("pg_config") is None:
+        print(f"pg_config not found", file=sys.stderr)
+        exit(1)
+    proc = subprocess.run(f"pg_config --sharedir", check=True, shell=True, env=os.environ, text=True,
+                          capture_output=True)
+    return Path(str(proc.stdout).strip()).resolve().absolute().joinpath("extension")
+
+
+def install_sql() -> None:
+    ext_dir = extension_dir()
+    if not ext_dir.exists():
+        print(f"extension directory does not exist: {ext_dir}", file=sys.stderr)
+        exit(1)
+    build_sql()
+    control_file = sql_dir().joinpath("ai.control")
+    shutil.copyfile(control_file, ext_dir.joinpath(control_file.name))
+    for src in sql_dir().glob("ai--*.sql"):
+        dest = ext_dir.joinpath(src.name)
+        shutil.copyfile(src, dest)
+
+
+def uninstall_sql() -> None:
+    ext_dir = extension_dir()
+    if not ext_dir.exists():
+        return
+    ext_dir.joinpath("ai.control").unlink(missing_ok=True)
+    for f in ext_dir.glob("ai--*.sql"):
```

Comment:
should this uninstall all ai files or only for this version? I am actually not sure what the right thing to do is.

Reply:
I don't know what the "right" thing is. Both options would be valuable. I intended it to be a "fully remove the pgai extension from my system" thing. For uninstall-py, I delete the directory with all the versions' python libs.

Maybe I should make `clean` also uninstall THIS version?